### PR TITLE
feat:Added org info API

### LINKED ISSUE #120: org info API
**Is your feature request related to a problem? Please describe.**
As a user, I should be able to see the info of an org.

**Describe the solution you'd like**
1. Create an api: `[get] /api/org/(org-id)/` which return org info.
1. only authenticated users can access this API.
1. make swagger docs for the same 
1. add tests for the same

**response**
```
{
    "id" : <id>  
    "route_slug" : <route_slug>  
    "can_join_without_invite" : <can_join_without_invite>    
    "name" : <name>  
    "tagline" : <tagline>  
    "about" : <about>   
    "profile_pic" : <profile_pic>   
    "cover_pic" : <cover_pic>  
}
```

### PR DESCRIPTION
**Pull request by sahil9001**
### Description
Added `org` info API which fetches the info about the organisation in which a user is present.

**Type of Change:** (Delete irrelevant options)
- Code
- Testing

**Code/Quality Assurance Only** (Delete irrelevant options)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.
```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
....................
......................................
----------------------------------------------------------------------
Ran 58 tests in 16.329s
```

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: org/urls.py [hunk 1]
```diff
@@ -4,15 +4,13 @@
 
 urlpatterns = [
     path('', OrgView.as_view()),
-    path('<int:org_id>/volunteer/',AddVolunteer,name='add_volunteer'),
-    path('<int:org_id>/group/', GroupView.as_view()),
-    path('<int:org_id>/', EditOrg, name='edit_org'),
+    path('<int:org_id>/', EditOrgView.as_view(), name='edit_org'),
     path('<int:org_id>/volunteer/', AddVolunteer, name='add_volunteer'),
+    path('<int:org_id>/group/', GroupView.as_view()),
     path('<int:org_id>/group/', GetGroup, name='get_group'),
```
Comment (naveennvrgup) [inline org/urls.py]:
```suggestion
```
line 9 already covers this case

File: org/urls.py [hunk 2]
```diff
@@ -4,15 +4,13 @@
 
 urlpatterns = [
     path('', OrgView.as_view()),
-    path('<int:org_id>/volunteer/',AddVolunteer,name='add_volunteer'),
-    path('<int:org_id>/group/', GroupView.as_view()),
-    path('<int:org_id>/', EditOrg, name='edit_org'),
+    path('<int:org_id>/', EditOrgView.as_view(), name='edit_org'),
```
Comment (naveennvrgup) [inline org/urls.py]:
```suggestion
    path('<int:org_id>/', OrgView.as_view(), name='edit_org'),
```
Please rename the class

File: org/views.py [hunk 3]
```diff
@@ -190,43 +188,80 @@ def AddVolunteer(request, org_id):
...
+    def get(self, request, org_id):
+        user = request.user
+        try:
+            org = Org.objects.get(pk=org_id)
+        except Org.DoesNotExist:
+            return Response(responses.org_not_present_400, status.HTTP_400_BAD_REQUEST)
+
+        try:
+            member = Member.objects.get(user=user, org=org)
+        except Member.DoesNotExist:
```
Comment (priyanshisharma) [inline org/views.py]: From what I interpret from the issue, any authenticated user could view organisation details and it doesn't necessarily need to be a member for the same. Could you please clarify @naveennvrgup sir?

File: org/views.py [hunk 4]
```diff
@@ -190,43 +188,80 @@ def AddVolunteer(request, org_id):
...
+            '200': set_example([{
+                "id": 1,
+                "route_slug": "slug",
+                "can_join_without_invite": True,
+                "name": "test",
+                "tagline": "test",
+                "about": "test",
+                "profile_pic":  "null",
+                "cover_pic":  "null"
+            }]),
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
            '200': set_example(responses.get_org_200),
```
Please extract it into responses.py

File: org/views.py [hunk 5]
```diff
@@ -190,43 +188,80 @@ def AddVolunteer(request, org_id):
...
+        user = request.user
+        try:
+            org = Org.objects.get(pk=org_id)
+        except Org.DoesNotExist:
+            return Response(responses.org_not_present_400, status.HTTP_400_BAD_REQUEST)
+
+        try:
+            member = Member.objects.get(user=user, org=org)
+        except Member.DoesNotExist:
+            return Response(responses.user_unauthorized_403, status.HTTP_403_FORBIDDEN)
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
```

Users need not be member of the org so the details. Like if the user want to see the details before join the org :thinking: .

File: org/views.py [hunk 6]
```diff
@@ -190,43 +188,80 @@ def AddVolunteer(request, org_id):
...
+
+        response_body = [{
+            "id": org.id,
+            "route_slug": org.route_slug,
+            "can_join_without_invite": org.can_join_without_invite,
+            "name": org.name,
+            "tagline": org.tagline,
+            "about": org.about,
+            "profile_pic": org.profile_pic if org.profile_pic else "null",
+            "cover_pic": org.cover_pic if org.cover_pic else "null"
```
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
            "profile_pic": org.profile_pic,
            "cover_pic": org.cover_pic
```
python's `None` will be automatically converted into javascripts `null`

File: tests/org/tests_get_group.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            self.org, self.admin_group = serializer.save()
+
+    def test_fail_without_auth(self):
+        get_org_api = "/api/org/1/"
+        un_auth_client = APIClient()
+        response = un_auth_client.get(get_org_api)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_fail_with_unknown_org(self):
+        get_org_api = "/api/org/2/"
```
Comment (naveennvrgup) [inline tests/org/tests_get_group.py]:
```suggestion
        get_org_api = "/api/org/12345/"
```
Use some really big numbers in case of invalid value. Just to be sure :wink:

File: tests/org/tests_get_group.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+from tests.AuthAPITestCase import AuthAPITestCase
+from rest_framework.test import APITestCase, APIClient
+from rest_framework import status
+from users.models import User
+from org.serializers import CreateOrgSerializer
+from org.custom_model_field import PermissionField as Permissions
+import uuid
+
+
+class EditOrgAPITestCase(AuthAPITestCase):
```
Comment (priyanshisharma) [inline tests/org/tests_get_group.py]: Also since this isn't related to groups I suppose the file name `tests_get_group.py`  must be renamed.

File: tests/org/tests_get_group.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_pass_with_auth_user(self):
+        get_org_api = "/api/org/1/"
+        auth_client = self.create_auth_client()
+        Member.objects.create(
+            user=self.auth_user,
+            org=self.org,
+            group=self.admin_group
+        )
```
Comment (naveennvrgup) [inline tests/org/tests_get_group.py]:
```suggestion
```
for the same reasons explained above

File: tests/org/tests_get_group.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        response = auth_client.get(get_org_api)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def tearDown(self):
+        self.auth_user.delete()
```
Comment (naveennvrgup) [inline tests/org/tests_get_group.py]:
```suggestion
```
1. AuthAPITestCase class automatically destroys this auth_user obj in its tearDown.
1. Since u didn't call the super class tearDown + u have overrided tearDown so self.un_auth_user wont be destroyed at all

**Review by naveennvrgup — CHANGES_REQUESTED** [review]

File: tests/org/tests_get_group.py (hunk 7 above)
Comment (priyanshisharma) [inline tests/org/tests_get_group.py]: Drawing this from @naveennvrgup sir at #110 , 

> Lets bring in the practice of using really large org ids for invalid orgs. So that minor changes like adding another org wont break the test cases.

```suggestion
        get_org_api = "/api/org/12345/"
```
It would be nicer to do it this way.

File: org/urls.py (hunk 1 above)
Comment (priyanshisharma) [inline org/urls.py]: Could you explain why did you add another path with `'int:org_id>/group/'` . I feel confused 😅

**Review by priyanshisharma — CHANGES_REQUESTED** [review]

**Comment by priyanshisharma** [comment]
Most of the suggestions I had got covered by @naveennvrgup sir a while ago, and so I marked them resolved, I suppose ignore them to avoid confusions. :)

File: org/urls.py (hunk 1 above)
Comment (sahil9001) [reply org/urls.py]: I see code duplication in the `upstream/dev` can you check it out?

File: org/views.py (hunk 6 above)
Comment (sahil9001) [reply org/views.py]: `ValueError: The 'profile_pic' attribute has no file associated with it.`

**Review by anshsrtv — CHANGES_REQUESTED** [review]
Screenshot from 2020-06-06 11-01-30
Check these URLs and solve it...

File: org/urls.py (hunk 1 above)
Comment (naveennvrgup) [reply org/urls.py]: Yup there is duplication. Which will be fixed with this commit. :tada:

**Comment by naveennvrgup** [comment]
Since this PR is beyond repair due to excessive unneed changes in `org/views.py` caused by pylint, this PR is being closed without merge.
